src/main/java/com/terraformersmc/terrestria/surface/RandomSurfaceConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.terraformersmc.terrestria.surface;
+
+import java.util.Random;
+
+import com.mojang.datafixers.Dynamic;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.world.gen.surfacebuilder.SurfaceConfig;
+
+public class RandomSurfaceConfig implements SurfaceConfig {
+
+	private final RandomNoiseFunction<BlockState> top;
+	private final RandomNoiseFunction<BlockState> under;
+	private final RandomNoiseFunction<BlockState> underwater;
+
+	public RandomSurfaceConfig(RandomNoiseFunction<BlockState> topStateProvider, RandomNoiseFunction<BlockState> underStateProvider, RandomNoiseFunction<BlockState> underwaterStateProvider) {
```

Comment:
As said before, surface configs should be data, not code. Maybe look at StateProvider as used in tree configs?